### PR TITLE
Make rpcs timeout aware so they cleanup correctly (new helpers)

### PR DESCRIPTION
**Pull request by chrisnojima**
This changes most of our rpcs to use a simplified flow.
99% of the time we would
1. Make a singleFixedChannelConfig
1. Make a channelMap
1. Do a race

This involved having to describe the keys in 2 places and to have to pull in a bunch of saga helpers. Now we can just use the generated code and pass in the channelConfig keys once.

This allows us to also: 
* Bake in timeout handling logic
* Remove so much knowledge of the saga helpers from actions
* Allows us to do more checking / unified logic
* Remove duplicate channelConfig map and race map
* Could let us flow type better (later)
* Better logging

This pr also:
- [x] Keeps track of dead sessions so we can output unhandled rpcs as entirely unknown or rpcs to cancelled/closed sessions
- [x] Marks timed out unboxed rows as untrusted so they can run again

Tested all the changed rpcs:
- [x] Login-Relogin
- [x] Login-Forgot password (removed, not referenced anywhere!)
- [x] Login-Logout
- [x] Login-Add device (Using old method, see note below)
- [x] Login-Login (Using old method, see note below)
- [x] Chat-inbox loading
- [x] Chat-thread loading
- [x] Chat-attachments (Using old method, see note below)
- [x] Device paperkey gen
- [x] Profile-generate pgp key
- [x] Profile-add service proof

We temporarily generate xxxRpcChannelMapOld versions of the old form so the login sagas and attachment sagas can be the same. Want to wait till @MarcoPolo is back to discuss. As these flows are doing things differently and I want to make sure they don't break. Once we decide what to do we can likely stop making the Old calls (and maybe some of the other forms as well).

### CODE REVIEW

File: shared/actions/devices.js [hunk 1]
```diff
@@ -32,6 +31,8 @@ const showRevokePage: (deviceID: string) => ShowRevokePage = deviceID => ({paylo
 
 const _loggedInSelector = (state: TypedState) => state.config.loggedIn
 
+const devicesTabLocation = isMobile ? [settingsTab, settingsDevicesTab] : [devicesTab]
```
Comment (chrisnojima) [inline shared/actions/devices.js]: This is to help navigation on desktop/mobile be more explicit. We had a bug where if you add a computer and cancel you'd end up in the loginscreen cause the login saga would try and send you to [devicesTab] on mobile.

File: shared/actions/login/creators.js [hunk 2]
```diff
@@ -132,10 +132,6 @@ function updateForgotPasswordEmail (email: string): Constants.UpdateForgotPasswo
   return {payload: email, type: Constants.actionUpdateForgotPasswordEmailAddress}
 }
 
-function submitForgotPassword () {
```
Comment (chrisnojima) [inline shared/actions/login/creators.js]: not used anywhere

File: shared/dev/dev-menu.native.js [hunk 3]
```diff
@@ -4,10 +4,9 @@ import React, {Component} from 'react'
 import engine from '../engine'
 import {connect} from 'react-redux'
 import {logout} from '../actions/login/creators'
-import {navigateAppend, switchTo} from '../actions/route-tree'
```
Comment (chrisnojima) [inline shared/dev/dev-menu.native.js]: i put this into a differetn pr, just found this while fixing this thing

File: shared/common-adapters/input.desktop.js [hunk 4]
```diff
@@ -152,7 +152,6 @@ class Input extends Component<void, Props, State> {
     ? {
       ...globalStyles.flexBoxRow,
       borderBottom: `1px solid ${underlineColor}`,
-      flex: 1,
```
Comment (chrisnojima) [inline shared/common-adapters/input.desktop.js]: I split this into a different PR

File: shared/profile/revoke/index.js [hunk 5]
```diff
@@ -63,7 +64,7 @@ const styleErrorBanner = {
 const styleErrorBannerText = {
   color: globalColors.white,
   maxWidth: 512,
-  textAlign: 'center',
+  ...(isMobile ? {} : {textAlign: 'center'}),
```
Comment (chrisnojima) [inline shared/profile/revoke/index.js]: pulled into a different pr

File: shared/actions/profile/proofs.js [hunk 6]
```diff
@@ -279,7 +271,7 @@ function * _submitCryptoAddress (action: SubmitBTCAddress | SubmitZcashAddress):
     yield call(cryptocurrencyRegisterAddressRpcPromise, {param: {address, force: true, wantedFamily}})
     yield put(_waitingForResponse(false))
     yield put(_updateProofStatus(true, ProveCommonProofStatus.ok))
-    yield put(navigateAppend(['postProof', 'confirmOrPending'], [profileTab]))
+    yield put(navigateAppend(['confirmOrPending'], [profileTab]))
```
Comment (chrisnojima) [inline shared/actions/profile/proofs.js]: don't put paths which we actually can't go to / shouldn't. on mobile since we render mulitple parts of the tree this actually crashes you out cause postProof doesn't allow certain types of things here (like btc)

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready for review

**Comment by cjb** [comment]
This isn't a review, but I've been using this branch today and haven't noticed any bugs in standard use so far.

**Comment by chrisnojima** [comment]
lets try and get this thing in. fixes bugs and we should let it bake
